File: deprecated/app/src/main/java/it/niedermann/nextcloud/deck/database/dao/JoinBoardWithUserDao.java
package it.niedermann.nextcloud.deck.database.dao;

import androidx.room.Dao;
import androidx.room.Query;

import it.niedermann.nextcloud.deck.model.JoinBoardWithUser;

@Dao
public interface JoinBoardWithUserDao extends GenericDao<JoinBoardWithUser> {
    @Query("DELETE FROM joinboardwithuser WHERE boardId = :localId")
    void deleteByBoardId(long localId);

}

File: data/local/src/main/java/it/niedermann/nextcloud/deck/data/local/dao/JoinBoardWithUserDao.kt
package it.niedermann.nextcloud.deck.data.local.dao

import androidx.room3.Dao
import androidx.room3.Query
import it.niedermann.nextcloud.deck.data.local.entity.JoinBoardWithUserEntity
import java.util.concurrent.CompletableFuture

@Dao
interface JoinBoardWithUserDao : GenericDao<JoinBoardWithUserEntity> {
    @Query("DELETE FROM JoinBoardWithUser WHERE boardId = :boardId")
    fun deleteByBoardId(boardId: Long): CompletableFuture<Void?>
}
